docs/exit_validation.md

```diff
@@ -14,52 +14,52 @@ This document focuses on the Watcher.
-    - because otherwise, unchallenged exits may break the chain without honest holders having time to escape.
-    This is the case when an exit, a spend invalidating it and exit of that coincide and are unchallenged.
-4. (optional) If a finalization of an invalid exit occurs, watcher must prompt an exit without guarantees of recovery of funds
+3. If an exit is invalid and remains unchallenged within a short period of time from its finalization, it must be challenged **AND** watcher must not allow spending and depositing, because:
+    a. we are running too close to the minimum safety requirements
+    b. unchallenged exits may break the chain without leaving honest holders with enough time to exit.
+    This can happen when an exit, a spend that invalidates that exit and an exit of that invalidating spend all coincide and are unchallenged.
+4. (optional) If finalization of an invalid exit occurs, the watcher must prompt an exit without guarantees of recovery of funds
 
 Conditions 3 and 4 can be represented jointly, by **continuous** (every child block) validation of the exits.
-If any exit is invalid and its finalization is near, then actions listed under 3 and 4 should take place
+If any exit is invalid and its finalization is near, then actions listed under 3 and 4 should take place.
 
 All that takes as an assumption that:
-  - the user's funds must be safe even if the user syncs and validates the chain periodically (but not less frequently than required)
-  - user needs to have the ability to constantly spend their UTXOs, thereby requiring more stringent validity checking of exits
+  - the user's funds must be safe even if the user only syncs and validates the chain periodically (but not less frequently than required)
+  - the user needs to have the ability to spend their UTXOs at any time, thereby requiring more stringent validity checking of exits
 
 ### Example cases for above actions:
-1. The exit is processed after the tx, so it is known to be invalid from the start. Should emit an `:invalid_exit` event.
-2. The exit is processed before the tx, but the tx is within `sla_margin` period, so the child chain is valid.
+1. The exit is processed *after* the tx that invalidates it, so it is known to be invalid from the start. Should emit an `:invalid_exit` event.
+2. The exit is processed *before* the tx that invalidates it and the tx is within `sla_margin` period, so the child chain is still valid.
 Causes an `:invalid_exit` event.
-3. The exit is processed before the tx, but tx is after `sla_margin` period, so the child chain is byzantine.
+3. The exit is processed *before* the tx that invalidates it, but the tx occurs *after* `sla_margin` period, so the child chain is byzantine.
 Causes to emit an `:unchallenged_exit` event.
-3. An exit is unchallenged within `sla_margin` period, so the child chain is jeopardized.
+3. An invalid exit is still unchallenged after `sla_margin` period, so the child chain is jeopardized.
```

Comment:
This is the opposite of what was there originally (i.e. the exit is unchallenged before the `sla_margin` is up). Have I misunderstood it?

Reply:
ah right, so this document follows the "end goal" approach, where exits become "unchallenged" when they enter a time period calculated based on finalization (`finalization - sla_margin`). So "still unchallenged within `sla_margin` from finalization" maybe?

But mind that implementation-wise, we're technically reaching this goal by calculating based on initiation (`startExit + sla_margin`). This is almost equivalent, and we have a future note for ourselves to amend that.

Regardless, I think the doc should state the former (`finalization - sla_margin`) consistently, If the new wording reads differently then this needs to be reverted

Reply:
Ok, makes sense but I think in some places it reads as if the sla_margin is counted ahead from when the exit is started.

I'll revise it again, but to be clear:
`sla_margin` is a number of ethereum blocks prior to finalization.

For example, if finalization is 10 blocks and sla_margin is 3 blocks, then it is possible to challenge an invalid exit for up to 7 blocks after it has started, but after that it becomes 'unchallenged' and the user should exit.

Correct?

Reply:
Yes, this is the intended behavior, currently implemented the converse way around.

Just be careful about a related thing from your comment:
It is still possible to challenge an invalid exit after `unchallenged`. Challenge must come before finalization. (Ah I see I worded that in a confusing way: `finalization` and `scheduled finalization time` could have been differentiated - Challenge must come before finalization, but we substract `sla_margin` from `scheduled finalization time (SFT)` c.f. Tesuji design doc).

It is just that after (whichever way around) some threshold moment had passed, exit is prompted, regardless of challenges etc.